### PR TITLE
Update logic for public.openTypePostUnderlinePosition key

### PR DESCRIPTION
**Pull request by benkiel**

### CODE REVIEW

**Comment by benkiel** [comment]
@ryanbugden does this make sense?

File: Underline Strikethrough.roboFontExt/lib/Underline StrikeThrough.py [hunk 1]
```diff
@@ -310,7 +310,19 @@ def updateFontList(self):
                             dictionary.update({fontIdentifier: value})
                         else:
                             dictionary.update({fontIdentifier: None})
-                        
+                # Deal with underline position
+                if 'public.openTypePostUnderlinePosition' in font.lib.keys():
+                    self.underlinePosition.update{fontIdentifier: font.lib['public.openTypePostUnderlinePosition'}
+                # If font doesn't have that lib key set, but is using fontmake, use fontmake behavior
+                elif font.lib["com.robofont.fontCompilerTool"] == 'fontmake':
```
Comment (ryanbugden) [inline Underline Strikethrough.roboFontExt/lib/Underline StrikeThrough.py]: @benkiel 
Do these two do the same thing?
```
from mojo.UI import getDefault
print(getDefault("fontCompilerTool"))

f = CurrentFont()
print(f.lib["com.robofont.fontCompilerTool"])
```
Got a `KeyError` for the 2nd. Maybe we should use `getDefault("fontCompilerTool")`?

Also:
What if someone has `fontmake` set as their compiler, but is preparing the font metadata for later generation with `afdko` through Terminal, or vice versa?

File: Underline Strikethrough.roboFontExt/lib/Underline StrikeThrough.py [hunk 2]
```diff
@@ -539,7 +551,12 @@ def setAllButtonCallback(self, sender):
         for font in self.fonts:
             fontIdentifier = self.getFontIdentifier(font)
             font.info.postscriptUnderlineThickness = self.roundInteger(uT[fontIdentifier])
-            font.info.postscriptUnderlinePosition  = self.roundInteger(uP[fontIdentifier])
+            # Math only works if there is a underline thickness
+            if uT[fontIdentifier]:
+                font.info.postscriptUnderlinePosition  = self.roundInteger(uP[fontIdentifier] - uT[fontIdentifier] / 2)
+            else:
+                font.info.postscriptUnderlinePosition = None
```
Comment (ryanbugden) [inline Underline Strikethrough.roboFontExt/lib/Underline StrikeThrough.py]: Not sure, but maybe would rather have some value? With no thickness, it's theoretically the same right?
```python
else:
    font.info.postscriptUnderlinePosition = self.roundInteger(uP[fontIdentifier])
```

File: Underline Strikethrough.roboFontExt/lib/Underline StrikeThrough.py [hunk 3]
```diff
@@ -310,7 +310,19 @@ def updateFontList(self):
                             dictionary.update({fontIdentifier: value})
                         else:
                             dictionary.update({fontIdentifier: None})
-                        
+                # Deal with underline position
+                if 'public.openTypePostUnderlinePosition' in font.lib.keys():
+                    self.underlinePosition.update{fontIdentifier: font.lib['public.openTypePostUnderlinePosition'}
+                # If font doesn't have that lib key set, but is using fontmake, use fontmake behavior
+                elif font.lib["com.robofont.fontCompilerTool"] == 'fontmake':
+                    self.underlinePosition.update{fontIdentifier: font.info.postscriptUnderlineThickness}
```
Comment (ryanbugden) [inline Underline Strikethrough.roboFontExt/lib/Underline StrikeThrough.py]: Should this be:
`self.underlinePosition.update{fontIdentifier: font.info.postscriptUnderlinePosition}`

File: Underline Strikethrough.roboFontExt/lib/Underline StrikeThrough.py (hunk 1 above)
Comment (ryanbugden) [reply Underline Strikethrough.roboFontExt/lib/Underline StrikeThrough.py]: I'm trying to think of other ways to intuit the intending meaning of the position. This is probably the best way, considering the default for RF is `afdko`, so someone would have had to manually choose `fontmake` as the compiler... 

... and ultimately this doesn't matter as long as the user mindfully uses the extension. Ignore this ramble lol...

Comment (benkiel) [reply Underline Strikethrough.roboFontExt/lib/Underline StrikeThrough.py]: Well, once ufo2ft is updated, it'll use the public key for `post` and you'll get the correct value

**Comment by ryanbugden** [comment]
🔥 Thanks @benkiel !

**Pull request merged**